storybookjs/storybook
# Indexer: Improve locating stories with specials chars in path

Closes #21636

## What I did

- Altered `StoryIndexGenerator.ts` not to treat `this.options.workingDir` as a glob. This allows the absolute part of the path to contain special glob characters without breaking the match.
- The problem and proposed solution are described inside of the issue.
- Note: As we're altering how we treat the local/absolute portion of the path, I'm unsure whether there is a way to test this.

## How to test

1. Insert a directory with a special character (e.g. `Special (1)`) in your path that contains a repository with a storybook.
3. Run the storybook linked to the `next` branch.
4. Verify that the stories are not loading (the storybook does not find any)
5. Run the storybook linked to `this` branch.
6. Verify that the stories are loaded.

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
